packages/wonder-blocks-dropdown/src/components/dropdown-core.tsx

```diff
@@ -585,7 +587,9 @@ class DropdownCore extends React.Component<Props, State> {
     }
 
     focusNextItem(): void {
-        if (this.focusedIndex === this.state.itemRefs.length - 1) {
+        if (this.isSearchFieldFocused() && !this.props.enableTypeAhead) {
+            this.focusedIndex = 0;
+        } else if (this.focusedIndex === this.state.itemRefs.length - 1) {
```

Comment:
Same suggestion here about combining both conditions into a single one:

```suggestion
    if (t(his.focusedIndex === this.state.itemRefs.length - 1) || (this.isSearchFieldFocused() && !this.props.enableTypeAhead)) {
```